flink-tests/src/test/java/org/apache/flink/test/recovery/JobManagerHAProcessFailureRecoveryITCase.java

```diff
@@ -325,6 +325,21 @@ public void run() {
 			dispatcherProcesses[1] = new DispatcherProcess(1, config);
 			dispatcherProcesses[1].startProcess();
 
+			// get new dispatcher gateway
+			leaderListener.waitForNewLeader(deadline.timeLeft().toMillis());
+
+			leaderAddress = leaderListener.getAddress();
+			leaderId = leaderListener.getLeaderSessionID();
+
+			final CompletableFuture<DispatcherGateway> newDispatcherGatewayFuture = rpcService.connect(
+				leaderAddress,
+				DispatcherId.fromUuid(leaderId),
+				DispatcherGateway.class);
+			final DispatcherGateway newDispatcherGateway = newDispatcherGatewayFuture.get();
+
+			// Wait for all task managers to connect to the new leading job manager
+			waitForTaskManagers(numberOfTaskManagers, newDispatcherGateway, deadline.timeLeft());
```

Comment:
Why do we have to add this code block? Shouldn't this implicitly be asserted if the jobs finishes successfully?

Reply:
Yes of course. I saw the removed test "A TaskManager detect a lost connection to the JobManager and try to reconnect to it" explicitly asserts this and thus add this block. I am not opposite to revert it since it is implicitly 
 asserted if the jobs finishes successfully.

Reply:
@tillrohrmann So I revert this change to proceed this pull request?

Reply:
Yes, I would be in favour of this.

Reply:
Done.

Reply:
Great @TisonKun.